services/app-api/src/resolvers/healthPlanPackage/updateHealthPlanFormData.ts

```diff
@@ -59,168 +114,277 @@ export function updateHealthPlanFormDataResolver(
+        if (ratesDatabaseRefactor) {
+            // Find contract from DB
+            const contractWithHistory = await store.findContractWithHistory(
+                input.pkgID
+            )
 
-        if (result === undefined) {
-            const errMessage = `No package found to update with that ID: ${input.pkgID}`
-            logError('updateHealthPlanFormData', errMessage)
-            setErrorAttributesOnActiveSpan(errMessage, span)
-            throw new UserInputError(errMessage, {
-                argumentName: 'pgkID',
-            })
-        }
+            if (contractWithHistory instanceof Error) {
+                const errMessage = `Issue finding a contract with history with id ${input.pkgID}. Message: ${contractWithHistory.message}`
+                logError('fetchHealthPlanPackage', errMessage)
+                setErrorAttributesOnActiveSpan(errMessage, span)
 
-        const planPackage: HealthPlanPackageType = result
+                if (contractWithHistory instanceof NotFoundError) {
+                    throw new GraphQLError(errMessage, {
+                        extensions: {
+                            code: 'NOT_FOUND',
+                            cause: 'DB_ERROR',
+                        },
+                    })
+                }
 
-        // Authorize the update
-        const stateFromCurrentUser = context.user.stateCode
-        if (planPackage.stateCode !== stateFromCurrentUser) {
-            logError(
-                'updateHealthPlanFormData',
-                'user not authorized to fetch data from a different state'
-            )
-            setErrorAttributesOnActiveSpan(
-                'user not authorized to fetch data from a different state',
-                span
-            )
-            throw new ForbiddenError(
-                'user not authorized to fetch data from a different state'
+                throw new GraphQLError(errMessage, {
+                    extensions: {
+                        code: 'INTERNAL_SERVER_ERROR',
+                        cause: 'DB_ERROR',
+                    },
+                })
+            }
+
+            // Authorize the update
+            const stateFromCurrentUser = context.user.stateCode
+            if (contractWithHistory.stateCode !== stateFromCurrentUser) {
+                logError(
+                    'updateHealthPlanFormData',
+                    'user not authorized to fetch data from a different state'
+                )
+                setErrorAttributesOnActiveSpan(
+                    'user not authorized to fetch data from a different state',
+                    span
+                )
+                throw new ForbiddenError(
+                    'user not authorized to fetch data from a different state'
+                )
+            }
+
+            // Can't update a submission that is locked or resubmitted
+            if (!['DRAFT', 'UNLOCKED'].includes(contractWithHistory.status)) {
```

Comment:
I wonder if this check is really necessary anymore now that we have the separate draftRevision slot

Reply:
Yes, that is true. However, I would opt to keep this in to have explicit errors for contracts that are not in draft and when they are in draft but have issues getting the draft revision.

Reply:
How is the 'DRAFT' status calculated? If it's calculated based on the presence of a draftRevision then this really feels redundant.

Reply:
Sounds good. Should I move the error message to the check below this? Like so:

```ts
// Can't update a submission that is locked or resubmitted
if (!contractWithHistory.draftRevision) {
    const errMessage = `Package is not in editable state: ${input.pkgID} status: ${contractWithHistory.status}`
    logError('updateHealthPlanFormData', errMessage)
    setErrorAttributesOnActiveSpan(errMessage, span)
    throw new UserInputError(errMessage, {
        argumentName: 'pkg',
    })
}
```

Because `findContractWithHistory` returns `ContractType` where `draftRevisions` can be undefined. We still need the check for now.

Reply:
yes that makes sense to me! I can imagine our future draft/submitted types making this check feel more secure